### ISSUE TITLE
Inquiry About Configuration Parameters and Performance Issues

### ISSUE THREAD

**Issue by Loveshuaiwei** [sunshangquan/TransMamba#7]
Thank you very much for your excellent work. I encountered some issues while attempting to reproduce the results of your paper. When training the model using the provided configuration file, the PSNR achieved on the Rain200H dataset was only around 30, which falls significantly short of the expected performance. Could you kindly share your YAML configuration files for both Rain200H and Rain200L for reference? This would greatly help me identify where the discrepancy might lie.
Here is my YAML configuration file for the Rain200H dataset:

# general settings
name: Deraining_TransMamba_Rain200H_train
model_type: ImageCleanModel
scale: 1
num_gpu: 2  # set num_gpu: 0 for cpu mode
manual_seed: 100

# dataset and data loader settings
datasets:
  train:
    name: TrainSet
    type: Dataset_PairedImage
    dataroot_gt: /home/ssw/dataset/Rain200H/train/target
    dataroot_lq: /home/ssw/dataset/Rain200H/train/input
    geometric_augs: true

    filename_tmpl: '{}'
    io_backend:
      type: disk

    # data loader
    use_shuffle: true
    num_worker_per_gpu: 8
    batch_size_per_gpu: 8

    ### -------------Progressive training--------------------------
    mini_batch_sizes: [8,5,2,1]             # Batch size per gpu   
    iters: [92000,88000,86000,84000]
    gt_size: 352   # Max patch size for progressive training
    gt_sizes: [128,160,224,352]  # Patch sizes for progressive training.
    ### ------------------------------------------------------------

    ### ------- Training on single fixed-patch size 128x128---------
    # mini_batch_sizes: [8]   
    # iters: [300000]
    # gt_size: 128   
    # gt_sizes: [128]
    ### ------------------------------------------------------------

    dataset_enlarge_ratio: 1
    prefetch_mode: ~

  val:
    name: ValSet
    type: Dataset_PairedImage
    dataroot_gt: /home/ssw/dataset/Rain200H/test/target
    dataroot_lq: /home/ssw/dataset/Rain200H/test/input
    io_backend:
      type: disk

# network structures
network_g:
  type: TransMamba
  inp_channels: 3
  out_channels: 3
  dim: 36
  num_blocks: [1,3,4,4]
  num_refinement_blocks: 2
  heads: [1,2,4,8]
  ffn_expansion_factor: 2.6667
  bias: False
  LayerNorm_type: WithBias
  dual_pixel_task: False

# path
path:
  pretrain_network_g: ~
  strict_load_g: true
  resume_state: ~

# training settings
train:
  total_iter: 350000
  warmup_iter: -1 # no warm up
  use_grad_clip: true

  # Split 300k iterations into two cycles. 
  # 1st cycle: fixed 3e-4 LR for 92k iters. 
  # 2nd cycle: cosine annealing (3e-4 to 1e-6) for 208k iters.
  scheduler:
    type: CosineAnnealingRestartCyclicLR
    periods: [92000, 258000]       
    restart_weights: [1,1]
    eta_mins: [0.0003,0.000001]   
  
  mixing_augs:
    mixup: false
    mixup_beta: 1.2
    use_identity: true

  optim_g:
    type: AdamW
    lr: !!float 3e-4
    weight_decay: !!float 1e-4
    betas: [0.9, 0.999]
    eps: !!float 1e-7
 
  # losses
  pixel_opt:
    type: L1Loss
    loss_weight: 1
    reduction: mean

  coherence_opt:
    type: CoherenceLoss
    loss_weight: 5
    reduction: mean

# validation settings
val:
  window_size: 8
  val_freq: !!float 1e3
  save_img: false
  rgb2bgr: true
  use_image: true
  max_minibatch: 8

  metrics:
    psnr: # metric name, can be arbitrary
      type: calculate_psnr
      crop_border: 0
      test_y_channel: true

# logging settings
logger:
  print_freq: 10
  save_checkpoint_freq: !!float 4e3
  use_tb_logger: true
  wandb:
    project: ~
    resume_id: ~

# dist training settings
dist_params:
  backend: nccl
  port: 29500

**Comment by sunshangquan** [comment]
Hi @Loveshuaiwei , thank you for your interest to our work. The YML should be
```
mini_batch_sizes: [8,5,2,1,1]             # Batch size per gpu   
iters: [92000,76000,48000,48000,36000]

gt_size: 368
gt_sizes: [128,160,256,320,368]

num_blocks: [2,4,6,8]
```

**Comment by Loveshuaiwei** [comment]
Thank you for your reply! I appreciate your help.
